modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer-fragment.glsl.js

```diff
@@ -36,23 +37,30 @@ varying vec2 uv;
 void main(void) {
   geometry.uv = uv;
 
-  vec4 texColor = texture2D(iconsTexture, vTextureCoords);
-  
-  float alpha = texColor.a;
+  float alpha = texture2D(iconsTexture, vTextureCoords).a;
+
   // if enable sdf (signed distance fields)
   if (sdf) {
-    float distance = texture2D(iconsTexture, vTextureCoords).a;
-    alpha = smoothstep(buffer - vGamma, buffer + vGamma, distance);
+    alpha = smoothstep(buffer - vGamma, buffer + vGamma, alpha);
   }
 
   // Take the global opacity and the alpha from vColor into account for the alpha component
   float a = alpha * vColor.a;
-
+  
   if (a < alphaCutoff) {
-    discard;
+    if ((backgroundColor.a < 0.001) || picking_uActive) {
+      discard;
+    } else {
+      gl_FragColor = vec4(backgroundColor.rgb, backgroundColor.a * vColor.a);
+      return;
+    }
   }
 
-  gl_FragColor = vec4(vColor.rgb, a);
+  if (backgroundColor.a < 0.001) {
+    gl_FragColor = vec4(vColor.rgb, a);
+  } else {
+    gl_FragColor = vec4(mix(backgroundColor.rgb, vColor.rgb, alpha), backgroundColor.a * vColor.a);
```

Comment:
See https://github.com/uber/luma.gl/blob/master/modules/shadertools/src/modules/picking/picking.js#L101 for how to do proper color blending.

Note that we don't blend colors in any other layer, e.g. between fill color and stroke color. I think it's fair and more efficient to just use `gl_FragColor = vec4(backgroundColor.rgb, backgroundColor.a * opacity)` here. If the user wants the color blended they can easily return a different instance color with `getColor`.

Reply:
This is a special case because the text is smooth (unlike other layers).
This is how it looks without blending:
<img width="324" alt="Screen Shot 2019-12-06 at 3 48 52 PM" src="https://user-images.githubusercontent.com/25012840/70364132-1ce58980-1840-11ea-93d2-f7662110490d.png">

Reply:
I see. In that case, the opacity should also be interpolated (`mix(backgroundColor.a, alpha, alpha) * opacity`). Multiplying the two alpha channels make the in-between fragments more translucent than either the text or the background which doesn't make sense.

Reply:
After playing with settings I think it is best to output constant opacity for both the text and background which will be `vColor.a`. This is basically the `opacity` property of the layer with no smoothing/interpolation in case of background.

Reply:
Okay can you update the documentation and the uniform generation then? You can make the prop default to `null` by adding `optional: true`

Reply:
OK, I'll do that.